coramin/relaxations/mccormick.py

```diff
@@ -158,3 +86,193 @@ def _build_mccormick_relaxation(b, x, y, w, relaxation_side=RelaxationSide.BOTH,
                 # see the doc string for this method - only adding one over and one under-estimator
                 if xlb != -math.inf and yub != math.inf:
                     b.relaxation.add(w <= x*yub + xlb*y - xlb*yub)
+
+
+def _build_pw_mccormick_relaxation(b, x, y, w, x_pts, relaxation_side=RelaxationSide.BOTH):
+    """
+    This function creates piecewise envelopes to relax "w = x*y". Note that the partitioning is done on "x" only.
+    This is the "nf4r" from Gounaris, Misener, and Floudas (2009).
+
+    Parameters
+    ----------
+    b: pyo.ConcreteModel or pyo.Block
+    x: pyomo.core.base.var._GeneralVarData
+        The "x" variable in x*y
+    y: pyomo.core.base.var._GeneralVarData
+        The "y" variable in x*y
+    w: pyomo.core.base.var._GeneralVarData
+        The "w" variable that is replacing x*y
```

Comment:
Elsewhere, you have moved to `aux_var` instead of `w`.  I think I like that convention, in that single-letter variables are harder to document / remember.

Reply:
Good catch. Thanks.